terraform/main.tf

```diff
@@ -0,0 +1,77 @@
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATE AN S3 BUCKET AND CLOUDFRONT DISTRIBUTION FOR THE GRUNTWORK.IO WEBSITE
+# ---------------------------------------------------------------------------------------------------------------------
+
+provider "aws" {
+  region              = "us-east-1"
+}
+
+terraform {
+  backend "s3" {
+    bucket         = "gruntwork-website-terraform-state"
+    key            = "terraform.tfstate"
+    region         = "us-east-1"
+    encrypt        = true
+    dynamodb_table = "terraform-locks"
+  }
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATE THE STATIC WEBSITE
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "static_website" {
+  source = "git::git@github.com:gruntwork-io/package-static-assets.git//modules/s3-static-website?ref=v0.0.1"
+
+  website_domain_name = "${var.domain_name}"
+
+  index_document = "${var.index_document}"
+  error_document = "${var.error_document}"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATE A REDIRECT BUCKET
+# This bucket just redirects non-www traffic to the www domain.
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "redirect" {
+  source = "git::git@github.com:gruntwork-io/package-static-assets.git//modules/s3-static-website?ref=v0.0.1"
+
+  website_domain_name          = "gruntwork.io"
+  should_redirect_all_requests = true
+  redirect_all_requests_to     = "${module.static_website.website_domain_name}"
+}
+
+# ---------------------------------------------------------------------------------------------------------------------
+# CREATE THE CLOUDFRONT WEB DISTRIBUTION
+# ---------------------------------------------------------------------------------------------------------------------
+
+module "cloudfront" {
+  source = "git::git@github.com:gruntwork-io/package-static-assets.git//modules/s3-cloudfront?ref=v0.0.1"
```

Comment:
I'm very happy to finally be dogfooding our own infra. It was wonderfully simple to get the S3 + CloudFront stuff running.